src/main/java/Command.java

```diff
@@ -0,0 +1,170 @@
+import java.io.IOException;
+
+class Command {
+
+    /**
+     * constructor.
+     * @param tasks is the list of tasks.
+     * @param ui is the UI class.
+     * @param storage is the Storage class.
+     * @throws DukeException when input is different from required.
+     * @throws IOException when input is different from required.
+     */
+    void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException, IOException{
+
+    }
+
+    /**
+     * whether is it an exit command
+     * @return true when it is an exit command, false otherwise.
+     */
+    boolean isExit() {
+        return (this instanceof ExitCommand);
+    }
+}
+
+class AddCommand extends Command {
```

Comment:
Please separate each class into a separate file. Each file should have one main class. If it's potentially messy, you can also package the different commands together as well.